src/tf/tf_parser.cpp

```diff
@@ -298,7 +298,7 @@ void tf_parser::parse_graph(const tensorflow::GraphDef& graph)
     {
         this->parse_node(p.first);
     }
-    auto last_ins = std::prev(mm->end());
+    auto last_ins = (mm->size() > 0) ? std::prev(mm->end()) : mm->end();
```

Comment:
You should do an early return here:

```cpp
if (mm->size() == 0)
    return;
```

Also the `if(last_ins != mm->end())` can be removed as well as its completely redundant.